### PR TITLE
Refactor MapGenerator.cpp into multiple files

### PR DESCRIPTION
**Pull request by LeftofZen**

### CODE REVIEW

File: src/OpenLoco/src/Map/MapGenerator.h [hunk 1]
```diff
@@ -1,19 +0,0 @@
-#pragma once
```
Comment (LeftofZen) [inline src/OpenLoco/src/Map/MapGenerator.h:1]: this is shown as 'deleted' but its just been moved into the subfolder

**Comment by AaronVanGeffen** [comment]
Thanks for the initiative on this. I think this makes it much easier to get an overview of the map generator logic.

I would prefer to keep `MapGenerator.{cpp,h}` in its current place, though, and instead create a folder for the height map generator files instead (`HeightMapGenerator`?). So pretty much what you've done, save for two files.

File: src/OpenLoco/src/Map/MapGenerator/HeightMap.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        {
+            return _height.data();
+        }
+
+        const uint8_t* data() const
+        {
+            return _height.data();
+        }
+
+        size_t size() const
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Map/MapGenerator/HeightMap.h:63]: Probably an `#include <cstddef>` will appease the Ubuntu CI.

Comment (ZehMatt) [reply src/OpenLoco/src/Map/MapGenerator/HeightMap.h:63]: Yeah, unfortunately C++ never directly said where to place it, one of those includes will do `<cstddef> <cstdio> <cstdlib>`

**Comment by ZehMatt** [comment]
> Thanks for the initiative on this. I think this makes it much easier to get an overview of the map generator logic.
> 
> I would prefer to keep `MapGenerator.{cpp,h}` in its current place, though, and instead create a folder for the height map generator files instead (`HeightMapGenerator`?). So pretty much what you've done, save for two files.

I think having the MapGenerator things in one folder is reasonable, the HeightMap is never used from anywhere else besides the map generators so this seems fine to me.

File: src/OpenLoco/src/Map/MapGenerator/HeightMap.h [hunk 3]
```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <cassert>
+#include <cstdint>
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Map/MapGenerator/HeightMap.h:4]:
```suggestion
#include <cstddef>
#include <cstdint>
```

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**